src/MCPClient/tests/test_create_dataverse_mets.py

```diff
@@ -0,0 +1,229 @@
+                        11, 3, 8, 2)
+
+
+class TestDataverseExample(object):
+    """Dataverse test runner class."""
+
+    write_dir = "fixtures/dataverse/dataverse_sources/dataverse_mets/"
+    fixture_path = "fixtures/dataverse/dataverse_sources"
+
+    THIS_DIR = os.path.dirname(os.path.abspath(__file__))
+    FIXTURES_DIR = os.path.join(THIS_DIR, fixture_path)
+    WRITE_DIR = os.path.join(THIS_DIR, write_dir)
+
+    mets_file_name = "METS.{}.dataverse.xml"
+
+    def setup_class(self):
+        """Initialize test output directory."""
+        try:
+            os.makedirs(self.WRITE_DIR)
+        except OSError as err:
+            print(
+                "Creating fixture directory failed: {}".format(err),
+                file=sys.stderr)
+
+    def teardown_class(self):
+        """Remove test output directory on completion of tests."""
+        shutil.rmtree(self.WRITE_DIR)
+
+    @pytest.mark.parametrize(
+        "fixture", [dv_1, dv_2, dv_3, dv_4, dv_5, dv_6, dv_7, dv_8])
+    def test_generated_mets_(self, fixture):
+        """Understand whether all the file and directory elements appear in the
+        METS as anticipated. A high-level test to quickly understand if
+        something has gone wrong while refactoring.
+        """
+        job = Job("stub", "stub", ["", ""])
+        convert_dataverse_structure.convert_dataverse_to_mets(
+            job=job, unit_path=self.FIXTURES_DIR,
+            dataset_md_name=fixture.fname, output_md_path=self.WRITE_DIR,
+            output_md_name=self.mets_file_name.format(fixture.fname))
+
+        mets_path = os.path.join(
+            self.WRITE_DIR, self.mets_file_name.format(fixture.fname))
+        if not os.path.isfile(mets_path):
+            pytest.fail("Fixtures were not previously setup correctly.")
+
+        try:
+            mets = metsrw.METSDocument.fromfile(mets_path)
+        except mets.MetsError:
+            pytest.fail("Could not parse mets {}".format(mets_path))
+
+        assert len(mets.all_files()) == \
+            fixture.all_file_count, (
+                "File count incorrect: '{}', expected '{}'"
+                .format(len(mets.all_files()), fixture.all_file_count))
+
+        dir_counter = 0
+        item_counter = 0
+        for file_ in mets.all_files():
+            if file_.type == "Directory":
+                dir_counter += 1
+            if file_.type == "Item":
+                item_counter += 1
+
+        assert dir_counter == \
+            fixture.dir_count, (
+                "Directory count incorrect: '{}', expected: '{}'"
+                .format(dir_counter, fixture.dir_count))
+        assert item_counter == \
+            fixture.item_count, (
+                "Item count incorrect: '{}', expected: '{}'"
+                .format(item_counter, fixture.item_count))
+
+    @pytest.mark.parametrize(
+        "fixture", [dv_1, dv_2, dv_3, dv_4, dv_5, dv_6, dv_7, dv_8])
+    def test_mets_content(self, fixture):
```

Comment:
Something I've noticed about this test is that it depends on the outcome of `test_generated_mets_`, e.g. the following command does not work but it should:

     docker-compose run --workdir /src/MCPClient --rm --user=root --entrypoint=py.test archivematica-mcp-client -x --reuse-db -vv /src/MCPClient/tests/test_create_dataverse_mets.py -k test_mets_content

Reply:
Here's a patch to your test file where `tmpdir` replaces `WRITE_DIR` and there's no shared state between tests: https://gist.github.com/sevein/2e224b41ba8f275a7096644c5c664044. With this you don't need `src/MCPClient/tests/fixtures/dataverse/dataverse_sources/dataverse_mets` to be ignored in `.gitignore`.